docs/semgrep-app/getting-started-with-semgrep-app.md

```diff
@@ -0,0 +1,284 @@
+---
+slug: getting-started-with-semgrep-app
+append_help_link: true
+title: Getting started with Semgrep App
+description: "Get started with Semgrep App to scan for security vulnerabilities on cloud repositories hosted on GitHub and GitLab."
+---
+
+# Scanning cloud repositories with Semgrep App
+
+Semgrep App enables you to run scans on multiple repositories by integrating with your GitHub or GitLab-hosted account. Scans are powered by **rules**. A Semgrep rule encapsulates pattern-matching logic and data-flow analysis used to find code violations, security issues, outdated libraries, and other issues.
+
+
+![Diagram of Semgrep App flow](../img/semgrep-app-diagram.png "Diagram of Semgrep App flow")
+
+
+Many rules are available from [Semgrep Registry](https://semgrep.dev/r), an open-source, community-driven repository of rules. Write your own rules to create general security checks through the Registry while customizing Semgrep to work for your team's specific secret-keeping and API practices.
+
+After performing a scan, use Semgrep App's Rule Board feature to prevent pull requests (PRs) or merge requests (MRs) from merging until findings are resolved. This helps prevent vulnerable code from shipping to widely-accessible environments, such as production or staging servers.

```

Comment:
Is the Rule Board important _before_ a scan? Specifically, you interact with the Rule Board before a scan to configure the behavior of future scans. The language here implies to me you scan, then decide, but in practice the workflow is the opposite.

Reply:
My assumptions about user flow went as follows:

(1) A complete Semgrep newbie wants to dive in and try out app (no prior experience with CLI)
(2) "I want to test Semgrep App" -> user adds a project
(3) Once project added, a scan **automatically** starts using the default ruleset (default behavior of the app.) A person entirely new to Semgrep and dipping their toes through app would not necessarily know Semgrep's "make rules then scan" flow.
(4) "I got good results, can I refine it further?" -> looks up rules (writing rules or rule board) OR
(5) "I didn't get great results", why? -> looks up rules (writing rules or rule board.)

I think my language is unclear here because I say "After performing a scan" - I should say, "After trying out Semgrep's default ruleset on some scans" or something similar.

What do you think?

Reply:
(Marking as resolved as there are no definite action items. Willl discuss with onboarding team these "assumptions".)